azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/Subnet.java

```diff
@@ -71,6 +93,7 @@
     interface Definition<ParentT> extends
         DefinitionStages.Blank<ParentT>,
         DefinitionStages.WithAddressPrefix<ParentT>,
+        DefinitionStages.WithNetworkSecurityGroup<ParentT>,
         DefinitionStages.WithAttach<ParentT> {
     }
 
```

Comment:
my reason for this is to be explicit about the intent, even though there may be duplication.
The intent for the Definition interface is to be a shortcut for the IMpl to implement *everything* that is needed to be implemented, regardless of whether it is optional or not.

The intent for the WithAttach is to explicitly identify the settings that are optional.
So the two intents for the two are different.

The approach you're proposing i think is interesting though, the difference is that the intent for Definition in that approach is to only identify the Required settings and rely on WithAttach/WithCreate to deal with the rest. Perhaps in fact this creates an opportunity for using a naming convention that makes this clearer:
Definition --> RequiredDefinition
WithAttach/WithCreate --> OptionalDefinition

Let me think about this some more...